joomla/joomla-cms
# [imp] Allow setting of Http status code in com_redirect

This feature allows the setting of the Http status code in com_redirect. This has many SEO related benefits as you may well wish to redirect with a non-301 code (a 501 or something crazy like that even).

It also gives much greater flexability to the redirect method in JApplication in order to do this. As that only allowed a redirect with a 301 or 303.

As this is an advanced option and should only be used by those who actually know what their doing I've dumped it in an advanced options parameter for the component. I've also built it in such a way that we can add more things into the advanced section in the future if we want. If the user has this turned off (which he will by default) then they will continue to redirect with a 301
## Testing Instructions

Apply the patch and also apply the database change found at https://github.com/wilsonge/joomla-cms/blob/redirect-upgrade/administrator/components/com_admin/sql/updates/mysql/3.4.0-2014-09-16.sql

Go into the backend and navigate to com_redirect (make sure you have the relevant plugin enabled).

Check that the component still works exactly the same as currently - we don't want to scare anyone off with new options :P 

Now we'll test the new feature - go into component options and enable the advanced mode (under advanced options tab). In the edit form for a given link this will give you the extra option to set a HTTP redirect number - this is any valid HTTP code. If you set a 3xx code then we redirect as usual. If you choose a non-3xx code then it will display the generic error page in Joomla with the selected error code.
